frontend/server/controllers/CourseController.php

```diff
@@ -1399,44 +1428,50 @@ public static function apiIntroDetails(Request $r) {
 
     /**
      * Returns course details common between admin & non-admin
-     * @param  Request $r
+     * @param Courses $course
+     * @param int $currentIdentityId
+     * @param bool $onlyIntroDetails
      * @return array
      */
-    private static function getCommonCourseDetails(Request $r, $onlyIntroDetails) {
+    private static function getCommonCourseDetails(
+        Courses $course,
+        int $currentIdentityId,
+        bool $onlyIntroDetails
+    ) : Array {
         $isAdmin = Authorization::isCourseAdmin(
-            $r['current_identity_id'],
-            $r['course']
+            $currentIdentityId,
+            $course
         );
 
         if ($onlyIntroDetails) {
             $result = [
                 'status' => 'ok',
-                'name' => $r['course']->name,
-                'description' => $r['course']->description,
-                'alias' => $r['course']->alias,
-                'basic_information_required' => $r['course']->needs_basic_information == '1',
-                'requests_user_information' => $r['course']->requests_user_information
+                'name' => $course->name,
+                'description' => $course->description,
+                'alias' => $course->alias,
+                'basic_information_required' => $course->needs_basic_information == '1',
```

Comment:
puedes cambiar todos los `== '1'` por un cast?

```suggestion
                'basic_information_required' => boolval($course->needs_basic_information),
```

Reply:
listo